spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/repository/support/SimpleSpannerRepository.java

```
@@ -129,4 +135,22 @@ public Iterable findAll(Sort sort) {
 	public Page findAll(Pageable pageable) {
 		return this.spannerOperations.findAll(this.entityType, pageable);
 	}
+
+	private <T> T doIfKey(Object key, Function<Key, T> operation) {
+		Key k;
+		boolean isIterable = Iterable.class.isAssignableFrom(key.getClass());
+		boolean isArray = Object[].class.isAssignableFrom(key.getClass());
+		if (isIterable || isArray) {
+			Key.Builder kb = Key.newBuilder();
+			for (Object keyPart : (isArray ? (Arrays.asList((Object[]) key))
+					: ((Iterable) key))) {
+				kb.appendObject(keyPart);
+			}
+			k = kb.build();
```

Should we throw an error when we pass in an empty array as a key?